src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -704,9 +709,43 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
         // expr has the default type if it appears in any other kind of statement
         case x if x.isInstanceOf[PStatement] => Some(DEFAULT_INTEGER_TYPE)
 
+        case e: PMisc => e match {
+          // The following case infers the type of an literal expression when it occurs inside a composite literal.
+          // For example, it infers that the expression `1/2` in `seq[perm]{ 1/2 }` has type perm. Notice that the whole
+          // expression would be parsed as
+          //   PCompositeLit(
+          //     PSequenceType(PPermissionType()),
+          //     PLiteralValue(Vector(
+          //       PKeyedElement(
+          //         None,
+          //         PExpCompositeVal(PDiv(PIntLit(BigInt(1)), PIntLit(BigInt(2))))))))
+          case comp if comp.isInstanceOf[PCompositeVal] => comp match {
```

Comment:
why didn't you use `case comp: PCompositeVal => ...`?

Reply:
out of distraction 😅